### PR TITLE
gtk4: update subclasses to the latest glib

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
cc @sdroege

### CODE REVIEW

File: gtk4/src/subclass/application.rs [hunk 1]
```diff
@@ -19,12 +18,12 @@ pub trait GtkApplicationImpl:
     }
 }
 
-pub trait GtkApplicationImplExt {
+pub trait ApplicationImplExt {
```
Comment (sdroege) [inline gtk4/src/subclass/application.rs:21]: Why did you rename the trait? This is annoying now because `gio::ApplicationImplExt` and `gtk::ApplicationImplExt` :)

File: gtk4/src/subclass/application.rs [hunk 2]
```diff
@@ -48,47 +47,34 @@ impl<T: GtkApplicationImpl> GtkApplicationImplExt for T {
...
-        unsafe extern "C" fn application_window_added<T: GtkApplicationImpl>(
-            ptr: *mut gtk_sys::GtkApplication,
-            wptr: *mut gtk_sys::GtkWindow,
-        ) {
-            let instance = &*(ptr as *mut T::Instance);
-            let imp = instance.get_impl();
-            let wrap: Borrowed<Application> = from_glib_borrow(ptr);
+unsafe impl<T: ApplicationImpl> IsSubclassable<T> for Application {
+    fn override_vfuncs(class: &mut ::glib::Class<Self>) {
+        <glib::Object as IsSubclassable<T>>::override_vfuncs(class);
```
Comment (sdroege) [inline gtk4/src/subclass/application.rs:52]:
```suggestion
        <gio::Application as IsSubclassable<T>>::override_vfuncs(class);
```

:scream:

Comment (bilelmoussaoui) [reply gtk4/src/subclass/application.rs:52]: Agh, my bad.

File: gtk4/src/subclass/application.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk4/src/subclass/application.rs:21]: I've followed what's done in gtk3 here

Comment (bilelmoussaoui) [reply gtk4/src/subclass/application.rs:21]: Ah no. I don't know what I was doing here. Fixed!

**Review by sdroege — APPROVED** [review]

File: gtk4/src/subclass/application.rs (hunk 2 above)
Comment (sdroege) [reply gtk4/src/subclass/application.rs]: Note that there's also some more magic happening there in gtk3, which I guess we'll include as port of going through all commits there and backporting as needed :)
